presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -186,15 +196,22 @@ public void close()
             return;
         }
         closed = true;
-        probe = null;
-        pageBuilder.reset();
+        reachedProbeEnd = true;
+        reset();
         // closing lookup source is only here for index join
         if (lookupSource != null) {
             lookupSource.close();
         }
         onClose.run();
     }
 
+    private boolean readyToBuildOutputPage()
```

Comment:
Agreed. However, additional argument for inlining is conflicts... I'm OK though we this as it is.